### PR TITLE
feat: `output_format`

### PR DESCRIPTION
**Pull request by DanielSchiavini**
### What I did
- Add output_format kwarg
- ~Add detect_vyper_version_from_source~
- Update tests
- Deletes the temporary file after compilation

Related issue: https://github.com/vyperlang/titanoboa/pull/294

### Checklist

- [x] I have confirmed that my PR passes all linting checks
- [x] I have included test cases
- [ ] I have updated the documentation (README.md)
- [x] I have added an entry to the changelog

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -28,95 +28,95 @@ jobs:
     - name: Run Tox
       run: tox -e lint
 
-  py36:
+  py37:
```
Comment (fubuloubu) [inline .github/workflows/main.yaml:31]: Python 3.7 is deprecated

Comment (DanielSchiavini) [reply .github/workflows/main.yaml]: Yeah :100: but I think it should be a separate PR to update all the dependencies

File: vvm/main.py [hunk 2]
```diff
@@ -23,13 +26,38 @@ def get_vyper_version() -> Version:
     return wrapper._get_vyper_version(vyper_binary)
 
 
+def detect_vyper_version_from_source(source_code: str) -> Optional[str]:
```
Comment (charles-cooper) [inline vvm/main.py:29]: i think separate this into a separate PR as it has separate scope

File: tests/conftest.py [hunk 3]
```diff
@@ -67,8 +67,12 @@ def all_versions(request):
 @pytest.fixture
 def foo_source(all_versions):
     visibility = "external" if all_versions >= Version("0.2.0") else "public"
+    interface = "IERC20" if all_versions >= Version("0.4.0a") else "ERC20"
+    import_path = "ethereum.ercs" if all_versions >= Version("0.4.0a") else "vyper.interfaces"
+    pragma_version = "pragma version" if all_versions >= Version("0.3.10") else "@version"
     yield f"""
-from vyper.interfaces import ERC20
+#{pragma_version} ^{all_versions}
```
Comment (charles-cooper) [inline tests/conftest.py:74]: this looks weird, `all_versions` is a weird name for the fixture. let's change it to `vyper_version`

File: tests/conftest.py [hunk 4]
```diff
@@ -77,16 +81,16 @@ def foo() -> int128:
 
 
 @pytest.fixture
-def foo_path(tmp_path_factory, foo_source, all_versions):
-    source = tmp_path_factory.getbasetemp().joinpath(f"Foo-{all_versions}.sol")
+def foo_path(tmp_path_factory, foo_source, vyper_version):
+    source = tmp_path_factory.getbasetemp().joinpath(f"Foo-{vyper_version}.sol")
```
Comment (charles-cooper) [inline tests/conftest.py:85]: `.sol`? haha

File: tests/test_compile_source.py [hunk 5]
```diff
@@ -30,3 +31,17 @@ def foo() -> int128:
     return 42
     """
     vvm.compile_source(source, vyper_version=version_str)
+
+
+def test_compile_metadata(foo_source, vyper_version):
+    if vyper_version <= Version("0.3.1"):
+        raise pytest.skip("metadata output not supported in vyper < 0.3.2")
```
Comment (charles-cooper) [inline tests/test_compile_source.py:38]: what does `raise pytest.skip` do?

**Review by charles-cooper — APPROVED** [review]

File: tests/test_compile_source.py (hunk 5 above)
Comment (charles-cooper) [reply tests/test_compile_source.py:38]: based on conversation in https://github.com/vyperlang/vvm/pull/22, please change this to regular `pytest.skip`

**Comment by DanielSchiavini** [comment]
@charles-cooper raw_compile is here: https://github.com/vyperlang/vvm/commit/e22fe43e55ec2fff5870c4e554be8fd8c21e427e
That might be more flexible. Let me know which version you prefer.

**Review by charles-cooper — APPROVED** [review]
